pkg/cloud/client.go

```diff
@@ -58,8 +58,8 @@ func NewClient(ccPath string) (Client, error) {
 		return nil, errors.Wrapf(err, "error encountered while parsing [Global] section from config at path: %s", ccPath)
 	}
 
-	c.cs = cloudstack.NewAsyncClient(cfg.APIURL, cfg.APIKey, cfg.SecretKey, cfg.VerifySSL)
-	c.csAsync = cloudstack.NewClient(cfg.APIURL, cfg.APIKey, cfg.SecretKey, cfg.VerifySSL)
+	c.csAsync = cloudstack.NewAsyncClient(cfg.APIURL, cfg.APIKey, cfg.SecretKey, cfg.VerifySSL)
```

Comment:
The client returned from NewAsyncClient works in a synchronous way. On the other hand, a client returned from NewClient works in an asychronous way. I know it's confusing but that's the way it is. Please revert back this change or we have to change all places using csAsync and cs.

Reply:
Got it - I wasn't sure about this so I'm glad you caught it. I'll leave a comment to that effect to avoid future confusion. Could you share a link to some documentation.

Reply:
https://pkg.go.dev/github.com/apache/cloudstack-go/v2/cloudstack#NewAsyncClient

It's annoying to me too.

> For sync API calls this client behaves exactly the same as a standard client call, but for async API calls this client will wait until the async job is finished or until the configured AsyncTimeout is reached. When the async job finishes successfully it will return actual object received from the API and nil, but when the timout is reached it will return the initial object containing the async job ID for the running job and a warning.